### PR TITLE
Review

### PR DESCRIPTION
**Pull request by xarvh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,62 @@
...
+
+
+playSound : SoundsDict String -> String -> Task.Task () ()
+playSound soundDictionary soundId =
+  case Dict.get soundId soundDictionary of
+    Just sound ->
+      Audio.playSound Audio.defaultPlaybackOptions sound
+
+    Nothing ->
+      Debug.crash <| soundId ++ " not loaded!"
```
Comment (eeue56) [inline README.md]: I wouldn't use Debug.crash here, just use `Task.fail "soundId not loaded"` and change the task to return a string on error

File: src/Native/Audio.js [hunk 2]
```diff
@@ -4,43 +4,64 @@ Elm.Native.Audio.make = function make(elm) {
...
-      return elm.Native.Audio.values;
+    return elm.Native.Audio.values;
   }
 
   var Task = Elm.Native.Task.make(elm);
 
 
   function loadSound(source) {
     return Task.asyncFunction(function (callback) {
+      if (typeof Audio === 'undefined') {
```
Comment (eeue56) [inline src/Native/Audio.js:10]: Note that `Audio` is also undefined on Node, so it will fail with elm-test too

File: src/Native/Audio.js [hunk 3]
```diff
@@ -4,43 +4,64 @@ Elm.Native.Audio.make = function make(elm) {
...
 
-  function playSound(options, sound) {
-//    for (var value, index = 0; value = options['_' + index]; index++)
-//        console.log('--', value.ctor);
 
+  function playSound(options, sound) {
     return Task.asyncFunction(function (callback) {
+      sound.audio.volume = options.volume;
+      sound.audio.loop = options.loop;
+      if (options.startAt.ctor == 'Just') { sound.audio.currentTime = options.startAt._0; }
```
Comment (eeue56) [inline src/Native/Audio.js]: `===`

File: src/Native/Audio.js [hunk 4]
```diff
@@ -4,43 +4,64 @@ Elm.Native.Audio.make = function make(elm) {
...
   function loadSound(source) {
     return Task.asyncFunction(function (callback) {
+      if (typeof Audio === 'undefined') {
+        return callback(Task.fail('The browser does not support HTML5 Audio'));
+      }
+
       var audio = new Audio();
+
+      function oncanplaythrough() {
+        callback(Task.succeed({ ctor: 'Sound', audio: audio }));
```
Comment (eeue56) [inline src/Native/Audio.js]: You can't toString this or eq it any more, but maybe you don't care about that

**Comment by eeue56** [comment]
Looks mostly good 👍

**Comment by xarvh** [comment]
Thanks, much appreciated.

**Pull request closed without merging**